### PR TITLE
[Storage] [Next-PyLint] Blobs Package

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR is the next-pylint work for Blobs.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py [hunk 1]
```diff
@@ -1293,10 +1298,6 @@ def _generate_delete_blobs_subrequest_options(
         modified_access_conditions=None,
         **kwargs
     ):
-        """This code is a copy from _generated.
-
-        Once Autorest is able to provide request preparation this code should be removed.
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:1298]: Can you turn this into just a regular comment. I think its good to note this but we don't need it in a docstring.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py [hunk 2]
```diff
@@ -175,10 +172,13 @@ def get_blob_properties_from_generated_code(generated):
...
-
-def parse_tags(generated_tags):
-    # type: (Optional[List[BlobTag]]) -> Union[Dict[str, str], None]
+# Deserialize a list of BlobTag objects into a dict.
+def parse_tags(generated_tags: Optional[List["BlobTag"]]) -> Union[Dict[str, str], None]:
     """Deserialize a list of BlobTag objects into a dict.
+    :param Optional[List[BlobTag]] generated_tags:
+            A list containing the BlobTag objects from generated code.
+        :returns: A dictionary of the BlobTag objects.
+        :rtype: Dict[str, str] or None
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py:181]: nit: indents
Also looks like you don't need the extra `# Deserialize a list of BlobTag objects into a dict.` on the top?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:1298]: Good catch. I remember nuking it, then in other files I turned it into just a regular comment but then I couldn't find the nuked sites. Will do!

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/parser.py [hunk 3]
```diff
@@ -23,18 +23,16 @@ def _str(value):
...
         return None
 
     return datetime.strptime(rfc_1123, "%a, %d %b %Y %H:%M:%S %Z")
 
+# Converts an MS filetime string to a UTC datetime. "0" indicates None.
+# If parsing MS Filetime fails, tries RFC 1123 as backup.
 def _filetime_to_datetime(filetime: str) -> datetime:
-    """Converts an MS filetime string to a UTC datetime. "0" indicates None.
-    If parsing MS Filetime fails, tries RFC 1123 as backup.
-    """
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/parser.py:37]: Can you fill these two in? They seem pretty straightforward

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py [hunk 4]
```diff
@@ -46,10 +41,10 @@ def serialize_iso(attr):
         return date + 'Z'
     except (ValueError, OverflowError) as err:
         msg = "Unable to serialize datetime object."
-        raise_with_traceback(ValueError, msg, err)
+        raise (ValueError, msg, err)  # pylint: disable=raising-bad-type
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py:44]: Not sure what this syntax is? Is that `raise` as a function? I think this is maybe some old Python 2 stuff?

Can we try changing to this? I think this should be functionally equivalent.
`raise ValueError("Unable to serialize datetime object.") from err`

I also think if you fix both of these you won't need to disable `inconsistent-return-statements` at the top of this function.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py [hunk 5]
```diff
@@ -175,10 +172,13 @@ def get_blob_properties_from_generated_code(generated):
...
     return blob
 
-
-def parse_tags(generated_tags):
-    # type: (Optional[List[BlobTag]]) -> Union[Dict[str, str], None]
+def parse_tags(generated_tags: Optional[List["BlobTag"]]) -> Union[Dict[str, str], None]:
     """Deserialize a list of BlobTag objects into a dict.
+
+    :param Optional[List[BlobTag]] generated_tags:
+    A list containing the BlobTag objects from generated code.
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_deserialize.py:179]: super nit: tab here

**Review by jalauzon-msft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py (hunk 4 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py:44]: So I think I originally tried that, and this is the result: 
`azure\storage\blob\_shared\request_handlers.py:44: [E0702(raising-bad-type), serialize_iso] Raising tuple while only classes or instances are allowed`

So I think we will have to just leave the disable in 😢

**Pull request merged**